File: php/features.php
<?php
declare(strict_types=1);

// WeatherDataApi3 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WeatherDataApi3Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WeatherDataApi3BaseFeature();
            case "test":
                return new WeatherDataApi3TestFeature();
            default:
                return new WeatherDataApi3BaseFeature();
        }
    }
}

File: py/features.py
# WeatherDataApi3 SDK feature factory

from feature.base_feature import WeatherDataApi3BaseFeature
from feature.test_feature import WeatherDataApi3TestFeature


def _make_feature(name):
    features = {
        "base": lambda: WeatherDataApi3BaseFeature(),
        "test": lambda: WeatherDataApi3TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
